Getting pug files up to coding standards ...

krishmaj opened the pull request:
- replaced non-CSS styling hacks with correct CSS styling - this should resolve issue #9
- moved previously embedded JavaScript code into new `/static/pages/` files
- moved previously embedded CSS styling into new `/static/css/` files
- cleaned up unnecessary whitespace and other formatting in pug files
- made code for determining page titles consistent for component, test and job list pages

Additional changes after 1st commit to resolve the following:
- removed `!important` CSS rule declarations which were being used in bad practice, and weren't actually doing anything anyway
- fixed bug with page titles not being shown in browser ... resolves issue #6

krishmaj committed:
Getting pug files up to coding standards ...
- replaced non-CSS styling hacks with correct CSS styling - this should resolve issue #9
- moved previously embedded JavaScript code into new /static/pages/ files
- moved previously embedded CSS styling into new /static/css/ files
- cleaned up unnecessary whitespace and other formatting in pug files
- made page titles consistent for component, test and job list pages

krishmaj commented on app/static/css/dashboard.css:
```diff
@@ -1,225 +1,182 @@
+    -o-animation     : none !important;
+    animation        : none !important;
+}
 
```
@micchickenburger I remember a while ago you brought up something about having these `!important` tags here ... can you remind me what that was about, and if it's something that we can resolve as part of this PR?

krishmaj replied:
Okay, so it actually looks like none of these `!important` tags are even needed - there are no other more specific rules in any of our CSS that relate to this `.disable-animations` class or any of its elements ... so nothing for the `!important` to even override.  Removing all of them makes no difference to the homepage (which is the only place where this class is used).

Will see about the other uses of this tag further down in the file.

krishmaj commented on app/static/css/dashboard.css:
```diff
@@ -1,225 +1,182 @@
-  box-shadow: inset -1px 0 0 rgba(0, 0, 0, .25);
+body.deployment-development #navbar-brand {
+    background-color: #fff3cd !important;
+    color           : black;
```
Same with these ones - they don't actually make any difference since there are no other more specific rules for these style elements that would be overridden.

krishmaj committed:
Removing 'important' CSS rule declarations, which are bad practice and aren't actually doing anything here anyway.

krishmaj committed:
Fixing bug with page titles not being displayed ... was simply that the title declaration had accidentally been moved inside the vars block.  Resolves issue #6.

krishmaj committed:
Some minor tweaks to wording of messages on the homepage.

krishmaj committed:
Small fix to user permission - logged in users can now view their own profile without needing additional permissions that not every user may have

krishmaj committed:
Small bug fix in common mixin - typo in variable name.

micchickenburger commented on app/pug/common.pug:
```diff
@@ -1,49 +1,17 @@
+block pugscripts
+  - function isAuthenticated()     { return !!(user); }
+  - function user_email(aUser)     { if (isAuthenticated()) return ((aUser.emails || [])[0] || {}).value || ""; else return "[Please login to see!]" };
+  - function user_shortname(aUser) { if (isAuthenticated()) return aUser.displayName || user_email(aUser); else return "[Please login to see!]" };
```
`else` is unneeded on lines 4 and 5.

krishmaj replied:
So by that you mean the syntax is `if (condition) return [value1]; return [value2]`?

micchickenburger replied:
Yup!

krishmaj replied:
Huh ... even after 6 months JavaScript continues to feel weird to me!

micchickenburger replied:
The same applies in C too.  The statement after the condition executes in all cases, except for the condition because there is a return statement inside of it.

micchickenburger commented on app/pug/common.pug:
```diff
@@ -1,49 +1,17 @@
 
+block pugscripts
+  - function isAuthenticated()     { return !!(user); }
```
This could be abbreviated:  `const isAuthenticated = () => !!user;`

krishmaj replied:
Is that some kind of JavaScript specific syntax?  Never seen something like that before.

micchickenburger replied:
They are called arrow functions and differ from regular functions only in local scope.

```javascript
// Traditional syntax
function isAthenticated() {}

// Function assignment
const isAuthenticated = function () {};

// Arrow function
const isAuthenticated = () => {};

// Arrow functions with one statement need no curly braces,
// and the return value of the statement becomes
// the return value of the function
const area = (radius) => Math.PI * (radius * radius);

// Functions can also be anonymous
(function () {})(); // traditional
(()=>{})(); // arrow
```

micchickenburger commented on app/pug/common.pug:
```diff
@@ -1,49 +1,17 @@
 
+block pugscripts
+  - function isAuthenticated()     { return !!(user); }
+  - function user_email(aUser)     { if (isAuthenticated()) return ((aUser.emails || [])[0] || {}).value || ""; else return "[Please login to see!]" };
```
I think this line is confusing and should have a comment describing what’s happening.

krishmaj replied:
I'll actually simplify the return statement to: `return (user.emails[0].value);` .  The current syntax assumes that a user might not have an associated email address, which used to be true with the old Sietch-related Auth0 tenant (never enforced requiring email addresses).  But with the new Auth0 setup this is never the case any more (since we now require users to have at least one).  Will still add a comment to describe the function anyway.

krishmaj committed:
Removing documents from DB ...
- removed static documentation from  directory, and associated pug rendering code
- removed ability to upload documents to the DB ... pug, routes, library, database

krishmaj committed:
Removing files from DB ...
- removed ability to add files to an entity form through Formio
- removed file routes

krishmaj committed:
Updating homepage and sidebar ...
- updated links in sidebar to point to the Github Wiki (for documentation) and CERN EDMS (for files)
- updated developer messages to also point to both external locations

micchickenburger commented on app/pug/splash.pug:
```diff
@@ -1,17 +1,15 @@
+  link(rel = 'stylesheet', href = '/dist/formiojs/formio.full.min.css')
+  link(rel = 'stylesheet', href = '/dist/bootstrap/css/bootstrap.min.css')
+  link(rel = 'stylesheet', href = 'https://fonts.googleapis.com/css2?family=Inconsolata&family=Palanquin&display=swap')
+  link(rel = 'preconnect', href = 'https://fonts.gstatic.com')
```
Question:  Is there benefit to this prefetch if the fonts are being loaded from a different domain?

krishmaj replied:
Honestly have no idea ... it was already like that, so I just left it as is, apart from cleaning up the code itself.  I can dive a bit deeper into it and get back to you.

krishmaj replied:
Looks like the `preconnect` line isn't needed, no ... as you say, the font stylesheet is already being loaded in the previous line anyway, and it doesn't appear to make any difference when removed.

There's also an identical situation in `dashboard.pug` - same two lines, with the `preconnect` one apparently making no difference at all.  I'll remove that one as well.

micchickenburger replied:
Preconnect statements enable browsers to establish connections to external domains before reading statements to load resources.  It can improve UX.  https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types/preconnect

krishmaj replied:
Right, I did read that page ... didn't fully understand the importance I guess.  So pointing the `preconnect` lines to `fonts.gstatic` doesn't do anything since that's not where the fonts are being loaded from ... they should point to `fonts.googleapis`?  And the `preconnect` line should go before that one, not after?

krishmaj replied:
Looking at it some more, would the `preload` link type be more appropriate here, instead of `preconnect`?  The MDN page for the former specifically uses the example of loading fonts from a CSS as an example of how using it can give performance improvement.

micchickenburger replied:
Sure, preload makes more sense I think.

micchickenburger commented on app/static/css/component.css:
```diff
@@ -0,0 +1,14 @@
+    padding         : 3px 3px;
+    border-radius   : 4px;
+    border          : 2px solid #008CBA;
+}
```
File needs a newline at the end

micchickenburger commented on app/static/css/component_qrCodes.css:
```diff
@@ -0,0 +1,155 @@
+    .qr-row {
+        page-break-inside: avoid;
+    }
+}
```
Missing newline at end of file

micchickenburger commented on app/static/css/component_summary.css:
```diff
@@ -0,0 +1,35 @@
+
+.section-header {
+    margin-left: 44px;
+}
```
Missing newline at end of file

micchickenburger commented on app/static/css/dashboard.css:
```diff
@@ -1,225 +1,182 @@
-  font-size: .75rem;
-  margin-right: 3px;
-  color: #999;
+    font-size   : .75rem;
```
This whole file:  Two spaces is de facto; colons should have no spaces to their left.

Proper:
```css
#sidebar .nav-link {
  font-size: 0.75rem;
}
```

krishmaj replied:
Sorry about that ... I think that the two-space thing (and the missing newlines) may have happened because I was editing the files using two different text editors, which have different space/tab and end-of-file settings.  Seen it happen before for the same reason.  Will fix.

micchickenburger commented on app/static/css/main.css:
```diff
@@ -45,19 +61,13 @@ pre {
+    border       : 3px solid red;
+    border-radius: 10px; 
+    padding      : 10px;
+}
```
Needs newline

micchickenburger commented on app/static/pages/component_summary.js:
```diff
@@ -0,0 +1,42 @@
+    
+    if(recordType === "component") { schema_url = "/json/componentForms/" + record.formId; }
+    else if(recordType === "test") { schema_url = "/json/testForms/" + record.formId; }
+    else if(recordType === "job")  { schema_url = "/json/jobForms/" + record.formId; }
```
Prefer backtick notation:
```schema_url = `/json/componentForms/${record.formId}`;```

krishmaj replied:
Should I change it everywhere in the code right now?  Pretty sure there's a lot more places where I've used `"" +` type notation ... everywhere where there's an `href` for example.

Or I can change it here specifically right now, and make a new issue to do the same changes across the rest of the code.

micchickenburger replied:
Just for new changes is fine for now I think.  We’ll automate these changes when we introduce linting.

micchickenburger commented on app/static/pages/edit_component.js:
```diff
@@ -0,0 +1,207 @@
+{
+  $('#loadRecentData').hide();
+  
+  $.post('/json/search/component/' + (component.formId || "") + "?limit=1", {"insertion.user.user_id": "!{user.user_id}"})
```
Prefer backtick notation for readability

micchickenburger reviewed (changes requested):
In all the CSS files it looks like you inadvertently changed the spacing from two to four spaces.  Also, colons should abut their property names.  Finally, newlines should exist in all files.

krishmaj committed:
Merge pull request #22 from DUNE/docsfiles

Removing documents and files from DB

krishmaj committed:
Partially addressing PR changes ...
- fixing formatting in CSS files - two spaces instead of four, newlines at EOF, and removed additional whitespace before colons
- fixing common pugscripts - simplified authentication check, simplified user email function, and added comments to describe each function

Waiting on additional feedback before submitting remaining requested changes.

krishmaj committed:
Addressing remaining PR changes ...
- removed preconnect links to fonts.gstatic.com ... don't appear to be doing anything, since the font stylesheets are being loaded from elsewhere anyway
- changed a couple of specific instances of 'quote + plus' notation to 'backtick' notation ... all other instances will be dealt with using linting

krishmaj committed:
Changing preconnect link types to preload, which make more sense for loading fonts from css.  Also removed the link to Markdown stylesheet, since that was only being used by documentation creation, which is no longer in the DB.
